core/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -2178,4 +2181,105 @@ public void testRollGenerationBetweenPrepareCommitAndCommit() throws IOException
 
     }
 
+    public void testMinGenerationForSeqNo() throws IOException {
+        final int operations = randomIntBetween(1, 4096);
+        final List<Long> seqNos =
+                LongStream.range(0, operations).boxed().collect(Collectors.toList());
+        Randomness.shuffle(seqNos);
+        final Map<Long, List<Long>> generations = new HashMap<>();
+
+        for (int i = 0; i < operations; i++) {
+            final Long seqNo = seqNos.get(i);
+            translog.add(new Translog.NoOp(seqNo, 0, "test"));
+            final List<Long> seqNoForGeneration =
+                    generations.computeIfAbsent(
+                            translog.currentFileGeneration(),
+                            g -> new ArrayList<>());
+            seqNoForGeneration.add(seqNo);
+            if (rarely()) {
+                translog.rollGeneration();
+            }
+        }
+
+        final Set<Long> seenSeqNos = new HashSet<>();
+        for (final Map.Entry<Long, List<Long>> entry : generations.entrySet()) {
+            final long min = entry.getValue().stream().min(Long::compareTo).orElse(Long.MIN_VALUE);
+            final long max = entry.getValue().stream().max(Long::compareTo).orElse(Long.MAX_VALUE);
+            for (long seqNo = min; seqNo <= max; seqNo++) {
+                if (seenSeqNos.add(seqNo)) {
+                    assertThat(
+                            translog.getMinGenerationForSeqNo(seqNo).translogFileGeneration,
```

Comment:
I'm confused - 

say we have:

```
gen 1 -> [1, 2, 8]
gen 2 -> [4, 5, 6]
gen 3 -> [7, 9]
```

wouldn't `translog.getMinGenerationForSeqNo(5)` be 1? if read this correctly it will expect it to be 2 (where `5` was indexed)